src/pages/popup/components/ConnectWalletForm.tsx

```diff
@@ -29,14 +29,15 @@ interface Inputs {
   autoKeyAddConsent: boolean;
 }
 
+type ConnectTransientState = DeepReadonly<PopupTransientState['connect']>;
```

Comment:
just to validate my thoughts, `DeepReadonly` to prevent unintended mutations to connect state inside `ConnectWalletForm` component?